shared/fs/common/hocs.js

```diff
@@ -0,0 +1,49 @@
+// @flow
+import {branch, compose, connect, renderNothing, type Dispatch, type TypedState} from '../../util/container'
+import {isLinux, isMobile} from '../../constants/platform'
+import * as FsGen from '../../actions/fs-gen'
+import {navigateAppend} from '../../actions/route-tree'
+
+// SecurityPrefsPromptingHoc prompts user about going to security preferences
+// to allow the kext if needed. It prompts at most once per app-restart. Which
+// means for example, if the user goes to Files tab and gets an prompt. they
+// won't get prompted again in Settings, or after they click "Back" in the
+// prompt and go back to Files tab again. This is to avoid spamming the user.
+// We have a link in the Settings page so if the user wants they can still find
+// the instructions.
+const SecurityPrefsPromptingHoc = isMobile
```

Comment:
The alternative I could think of was to do this in the two containers, which didn't seem great since 1) it'd need to be duplicated; and 2) it'd be mixed with other logics actually for the components.

I actually find separating stuff into HoCs easier to understand/read than having larger containers, as they separate responsibilities properly. They don't cause re-rendering either once mounted, if I understand correctly.

Reply:
They're also yet another place you have to consult to understand what a component is doing (index, container, hoc).

> The alternative I could think of was to do this in the two containers, which didn't seem great since 1) it'd need to be duplicated; and 2) it'd be mixed with other logics actually for the components.

I'm confused--why can't the functionality be factored out into its own component?

Reply:
> I'm confused--why can't the functionality be factored out into its own component?

By its own component, did you mean where they need to `routeAppend`? That's what I was talking about; two separate components (Settings/Files, and Files) need this.